sprocket-doc.md

```diff
@@ -0,0 +1,79 @@
+
+## Custom themes
+
+While it is technically possible to supply your own custom CSS styling, this capability is currently undocumented. We recommend you stick with the default styling at this point in time, but do let us know what kinds of customization you would like to see in future releases! 
+
+## Using preamble comments for file-level documentation
+
+To provide top-level documentation for a file, add a comment block before the `version` statement where each line starts with `##`. These preamble comments will be rendered as Markdown above the generated table of contents on that file's dedicated page. For example:
+
+```wdl
+## # This is a header
+##
+## This is a paragraph with **bolding**, _italics_, and `code` formatting.
+
+version 1.2
+
+workflow foo {}
+```
+
+## v1.0 and v1.1 structs
+
+The WDL specification does not offer a way to document structs prior to WDL v1.2, so the pages for them are rather limited. The pages for these structs will have a copy of the raw WDL definition.
+
+## v1.2 structs
+
+Structs defined in v1.2 WDL _will_ eventually get rich HTML documentation similar to tasks and workflows. However this has not yet been implemented, so they are given the same treatment as pre-v1.2 structs.
+
+## Meta entries with special handling
+
+All meta entries will render in the final HTML documentation, but there are some special cases we introduce. Each of the below keys are expected to have a WDL `String` value.
+
+`description`
+: Every struct, task, and workflow `meta` section should have a `description` key. This description string can have Markdown formatting. The `description` string should be less than 140 characters or it will be clipped in some contexts.
+
+`help`
+: This text can be of any length. It's best practice to keep `description` short, and put any additional text needed under the `help` key. Help strings can also be styled with Markdown.
+
+`category`
+: Workflows can have a `category` key which will group workflow pages on the left sidebar.
+
+`external_help`
+: This key should have a URL as its value (i.e. a valid hyperlink represented as a WDL `String`), and will be rendered as a button which will open a new tab or window visiting the link.
+
+`warning`
+: This text will be rendered in a special "warning box" to draw the attention of users.
+
+## Parameters (inputs and outputs)
+
+Each input and output to a workflow or task should be documented, but there is some flexibility in the specifics. To get the most out of `sprocket doc`, it's recommended that each instance of parameter documentation be a meta object. That object should have at least a `description` key. If a parameter has a `String` value for its meta entry instead of a meta object, that string value will be treated as if it were the `description` key of a meta object with no other entries.
```

Comment:
```suggestion
Each input and output to a workflow or task should be documented, but there is some flexibility in the specifics. To get the most out of `sprocket doc`, it is recommended that each instance of parameter documentation be a meta object. That object should have at least a `description` key. If a parameter has a `String` value for its meta entry instead of a meta object, that string value will be treated as if it were the `description` key of a meta object with no other entries.
```

Reply:
Does this need an example? Or is "meta object" clear enough?

Reply:
My edit above is personal preference. I try to avoid contractions in formal writing, such as documentation.